### PR TITLE
TAJO-1739: Add a statement for adding partition to TajoDump

### PR DESCRIPTION
**Pull request by blrunner**
Add a statement for adding partition to TajoDump

### CODE REVIEW

File: tajo-catalog/tajo-catalog-common/src/main/proto/CatalogProtos.proto [hunk 1]
```diff
@@ -407,3 +407,8 @@ message IndexResponse {
   required ReturnState state = 1;
   optional IndexDescProto indexDesc = 2;
 }
+
+message GetPartitionsResponse {
```
Comment (jihoonson) [inline tajo-catalog/tajo-catalog-common/src/main/proto/CatalogProtos.proto]: Every message for rpc communication with catalog should locate in CatalogProtocol.proto.

Comment (jihoonson) [inline tajo-catalog/tajo-catalog-common/src/main/proto/CatalogProtos.proto]: You seem to move this message to here to use in TajoMasterClientProtocol. If so, rename it like other messages.

File: tajo-client/src/main/java/org/apache/tajo/client/CatalogAdminClientImpl.java [hunk 2]
```diff
@@ -241,6 +241,22 @@ public TableDesc getTableDesc(final String tableName) throws UndefinedTableExcep
...
+
+    GetPartitionsResponse response;
+    try {
+      response = stub.getPartitionsByTableName(null,
+        conn.getSessionedString(tableName));
+    } catch (ServiceException e) {
+      throw new RuntimeException(e);
+    }
+
+    ensureOk(response.getState());
```
Comment (jihoonson) [inline tajo-client/src/main/java/org/apache/tajo/client/CatalogAdminClientImpl.java]: ```UndefinedTableException``` and ```UndefinedDatabaseException``` must be handled.

**Comment by blrunner** [comment]
@jihoonson 

I reflected your comments.

**Comment by jihoonson** [comment]
I missed that AbstractDBStore.getPartitions() already throws UndefinedTableException and UndefinedDatabaseException properly. Sorry for the last comment.

**Comment by jihoonson** [comment]
+1 looks good to me!

**Comment by blrunner** [comment]
Thanks @jihoonson 
I'll ship it. :)

**Pull request closed without merging**